pkg/controller/secretannotator/vsphere/reconciler.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2020 The OpenShift Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package vsphere
+
+import (
+	"context"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+
+	minterv1 "github.com/openshift/cloud-credential-operator/pkg/apis/cloudcredential/v1"
+	"github.com/openshift/cloud-credential-operator/pkg/controller/metrics"
+	"github.com/openshift/cloud-credential-operator/pkg/controller/secretannotator/constants"
+	"github.com/openshift/cloud-credential-operator/pkg/controller/utils"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	controllerName = "secretannotator"
+
+	// VSphereCloudCredSecretName is the name of the secret where credentials
+	// for vSphere are stored.
+	VSphereCloudCredSecretName = "vsphere-creds"
```

Comment:
Is it safe to assume this hard-coded value instead of getting the secret name from the cloud provider config? What if there are multiple vCenters with different cred secrets?

Reply:
this is an excellent question. @jcpowermac do you know if it is safe to assume that the vsphere secret will always be the secret kube-system/vsphere-creds ?

Reply:
I haven't tested it but I wonder if you could continue to append the data section:
```
vcenterA.username: A
vcenterA.password: A
vcenterB.username: B
vcenterB.password: B
```
https://vmware.github.io/vsphere-storage-for-kubernetes/documentation/k8s-secret.html

Reply:
@dav1x has done a lot with it maybe he has additional insight.

Reply:
the secret does have a format that would seem to allow this
```
data:
  vcsa-ci.vmware.devcluster.openshift.com.password: some_base64_password
  vcsa-ci.vmware.devcluster.openshift.com.username: some_base64_username
```

Reply:
It was only recently that VCP started to support multiple credentials or secrets via separate vCenter servers. 

https://github.com/kubernetes/cloud-provider-vsphere/pull/241

The examples in the PR above do a good job illustrating how this should look.